implementations/micrometer-registry-humio/src/test/java/io/micrometer/humio/HumioMeterRegistryTest.java

```diff
@@ -52,7 +69,83 @@ void datasourceTags(@WiremockResolver.Wiremock WireMockServer server) {
                 .withRequestBody(containing("\"tags\":{\"name\": \"micrometer\"}")));
     }
 
-    private HumioMeterRegistry humioRegistry(WireMockServer server, String... tags) {
+    @Test
+    void writeGauge() {
+        meterRegistry.gauge("my.gauge", 1d);
+        Gauge gauge = meterRegistry.find("my.gauge").gauge();
+        assertThat(createBatch().writeGauge(gauge)).isNotNull();
+    }
+
+    private HumioMeterRegistry.Batch createBatch() {
+        return meterRegistry.new Batch(clock.wallTime());
+    }
+
+    @Test
+    void writeGaugeShouldDropNanValue() {
+        meterRegistry.gauge("my.gauge", Double.NaN);
+        Gauge gauge = meterRegistry.find("my.gauge").gauge();
+        assertThat(createBatch().writeGauge(gauge)).isNull();
+    }
+
+    @Test
+    void writeGaugeShouldDropInfiniteValues() {
+        meterRegistry.gauge("my.gauge", Double.POSITIVE_INFINITY);
+        Gauge gauge = meterRegistry.find("my.gauge").gauge();
+        assertThat(createBatch().writeGauge(gauge)).isNull();
+
+        meterRegistry.gauge("my.gauge", Double.NEGATIVE_INFINITY);
```

Comment:
Not a big deal, but ideally this should be represented as 2 independent test cases - for negative and positive infitity as per unit test conventions